app/src/main/java/com/infomaniak/mail/data/models/calendar/CalendarEvent.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+@file:UseSerializers(RealmListKSerializer::class, CalendarRealmInstantSerializer::class)
+
+package com.infomaniak.mail.data.models.calendar
+
+import com.infomaniak.mail.data.api.CalendarRealmInstantSerializer
+import com.infomaniak.mail.utils.toRealmInstant
+import io.realm.kotlin.ext.realmListOf
+import io.realm.kotlin.serializers.RealmListKSerializer
+import io.realm.kotlin.types.EmbeddedRealmObject
+import io.realm.kotlin.types.RealmInstant
+import io.realm.kotlin.types.RealmList
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+import java.util.Date
+
+@Serializable
+class CalendarEvent : EmbeddedRealmObject {
+
+    //region Remote data
+    var type: String = ""
+    var title: String = ""
+    var description: String = ""
+    var location: String? = null
+    @SerialName("fullday")
+    var isFullDay: Boolean = false
+    var start: RealmInstant = Date(0).toRealmInstant()
+    var end: RealmInstant = Date(0).toRealmInstant()
+    var timezone: String? = null
+    @SerialName("timezone_start")
+    private var timezoneStart: String = ""
+    @SerialName("timezone_end")
+    private var timezoneEnd: String = ""
+    var attendees: RealmList<Attendee> = realmListOf()
+    //endregion
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as CalendarEvent
+
+        if (type != other.type) return false
+        if (title != other.title) return false
+        if (description != other.description) return false
+        if (location != other.location) return false
+        if (isFullDay != other.isFullDay) return false
+        if (start != other.start) return false
+        if (end != other.end) return false
+        if (timezone != other.timezone) return false
+        if (timezoneStart != other.timezoneStart) return false
+        if (timezoneEnd != other.timezoneEnd) return false
+        return attendees == other.attendees
```

Comment:
You mean returning on every line instead of computing a big single boolean with a lot of "and" operator?

If so, this is mainly here that way because that's how Android Studio generates it which is the easiest way of implementing it rather than writing it all by yourself. I don't know if this way of writing it brings any kind of advantage in performance or readability

Reply:
Well I don't know if there's compiling improvement that takes off the ifs, but if it's not the case then we add operation and jump for each if so that wouldn't be optimal compared to boolean operation

Reply:
Jumps happen in both cases anyway. We added a comment to try and warn future devs to not update the fields without adding them to the equals at least